xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/XWiki.java

```diff
@@ -2409,22 +2420,29 @@ public String getSkinFile(String filename, boolean forceSkinAction, XWikiContext
             if (skin != null) {
                 Resource<?> resource = skin.getResource(filename);
                 if (resource != null) {
-                    return resource.getURL(forceSkinAction);
+                    queryParameters.putAll(getResourceURLParameters(resource));
```

Comment:
You're modifying an input parameter which means this method has side effects that are probably not expected from a "getter". Moreover, the ``queryParameters`` map cold be unmodifiable.

Reply:
> You're modifying an input parameter which means this method has side effects that are probably not expected from a "getter"

Agreed but here the method is in fact about creating a skinFile URL, so from the beginning it's not a "real" getter IMO. Might be even renamed "createSkinFileURL". 

> Moreover, the queryParameters map cold be unmodifiable.

Right I should create a new map.